typify-impl/src/type_entry.rs

```diff
@@ -867,9 +867,31 @@ impl TypeEntry {
                 let value_output = enum_values
                     .iter()
                     .map(|value| sub_type.output_value(type_space, &value.0));
+
+                // If the sub type is a new type, than we additionally need to implement `PartialEq`
+                // to ensure that the `TryFrom` below is correct. We must make sure that we only
+                // implement `PartialEq` here for types that we define. Primitive types and Vecs
+                // already have implementations.
+                let p_eq = if let TypeEntry {
```

Comment:
I think I led you astray. I don't think we should generate impls for types other than the one we're processing. Can we add this to the case below that starts
```rust
            TypeEntryNewtypeConstraints::String {
                max_length,
                min_length,
                pattern,
            } => {
```